File: execution/analyze_selectors.py
import os, re
from collections import Counter

matches = []
for root, dirs, files in os.walk('src/content'):
    for file in files:
        if file.endswith('.js'):
            with open(os.path.join(root, file), 'r', encoding='utf-8') as f:
                content = f.read()
                found = re.findall(r"querySelector\(['\"](.*?)['\"]\)", content)
                matches.extend(found)
                found_all = re.findall(r"querySelectorAll\(['\"](.*?)['\"]\)", content)
                matches.extend(found_all)

c = Counter(matches)
for k, v in c.most_common(20):
    print(f"{v}: {k}")

File: scripts/execution/analyze_selectors.js
const fs = require('fs');
const path = require('path');

function walk(dir) {
    let results = [];
    const list = fs.readdirSync(dir);
    list.forEach(function(file) {
        file = path.join(dir, file);
        const stat = fs.statSync(file);
        if (stat && stat.isDirectory()) { 
            results = results.concat(walk(file));
        } else { 
            if(file.endsWith('.js')) {
                results.push(file);
            }
        }
    });
    return results;
}

const files = walk('src/content');
let matchCount = {};

for(const file of files) {
    let content = fs.readFileSync(file, 'utf8');
    
    // findall querySelector/querySelectorAll
    const regex = /querySelector(?:All)?\(['"`](.*?)['"`]\)/g;
    let match;
    while ((match = regex.exec(content)) !== null) {
        const sel = match[1];
        matchCount[sel] = (matchCount[sel] || 0) + 1;
    }
}

const sorted = Object.entries(matchCount).sort((a, b) => b[1] - a[1]);
for(let i = 0; i < 30; i++) {
    console.log(`${sorted[i][1]}: ${sorted[i][0]}`);
}
